internal/model/error.go

```diff
@@ -0,0 +1,60 @@
+package model
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+)
+
+type validationError struct {
+	message   string
+	subErrors []error
+}
+
+func newValidationError(message string) *validationError {
+	return &validationError{message: message}
+}
+
+func (v *validationError) orNil() error {
+	if len(v.subErrors) == 0 {
+		return nil
+	}
+	return v
+}
+
+func (v *validationError) Error() string {
+	if v == nil {
+		return ""
```

Comment:
This acts strange when the leaf subErrors is empty, validationError probably needs to either not print an empty error or add a newline. An empty leaf gives an error that looks like
```root:
├── foo└── bar:
    └── baz
```
instead of 
```root:
├── foo
└── bar:
    └── baz
```

Reply:
Fixed and added test